nilearn/surface/surface.py

```diff
@@ -619,27 +619,51 @@ def _gifti_img_to_mesh(gifti_img):
             coords = gifti_img.get_arrays_from_intent(
                 nibabel.nifti1.intent_codes['NIFTI_INTENT_POINTSET'])[0].data
         except IndexError:
-            raise ValueError('Gifti file needs to contain a data array '
-                             'with intent NIFTI_INTENT_POINTSET')
+            raise ValueError('The surf_mesh input is not recognized.'
```

Comment:
As the same message is repeated multiple times we could write like this, **above** nibabel version check, something like this:

```python
error_message = ('The surf_mesh input is not recognized. Valid Freesurfer '
                 'surface mesh inputs are .pial, .inflated, .sphere, '
                 '.orig, .white. You provided input which has no '
                 '{0} or of empty value={1}')    

if LooseVersion(nibabel.__version__) >= LooseVersion('2.1.0'): 
```
 
Try to raise error message like this at relevant place pointing to NIFTI_INTENT_POINTSET:

```python
raise ValueError(error_message.format(
    'NIFTI_INTENT_POINTSET', gifti_img.get_arrays_from_intent(
    nibabel.nifti1.intent_codes['NIFTI_INTENT_POINTSET'])))
```


Like this pointing to NIFTI_INTENT_TRIANGLE
```python
raise ValueError(error_message.format(
    'NIFTI_INTENT_TRIANGLE', gifti_img.get_arrays_from_intent(
    nibabel.nifti1.intent_codes['NIFTI_INTENT_TRIANGLE'])))
```

Reply:
Is it efficient  to assign a variable just to print an error message  but if you're saying it I will do it

Reply:
Just to make it look clean. I don't better alternative. If you have you can propose.

Reply:
@himanshupathak21061998 If we don't assign a variable, then the information/value is assigned to an invisible, temporary variable in the background; the memory consumption is nearly the same, explicit variable or not.
Assigning it to a variable makes the code easier to parse for anyone who works with the code regularly and is considered a "Clean code" practice, industry-wide.

Reply:
OK I  will change it for a clean code thanks for suggestion .